File: app/src/main/java/info/hannes/liveedgedetection/demo/SplashScreen.java
package info.hannes.liveedgedetection.demo;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;

public class SplashScreen extends AppCompatActivity {
    private static final int SPLASH_DISPLAY_TIME = 1000;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);

        new Handler().postDelayed(new Runnable() {
            public void run() {

                Intent intent = new Intent(SplashScreen.this, StartActivity.class);
                startActivity(intent);
                finish();

            }
        }, SPLASH_DISPLAY_TIME);
    }
}

File: app/src/main/java/info/hannes/liveedgedetection/demo/SplashScreen.kt
package info.hannes.liveedgedetection.demo

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity

class SplashScreen : AppCompatActivity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)
        Handler().postDelayed({
            val intent = Intent(this@SplashScreen, StartActivity::class.java)
            startActivity(intent)
            finish()
        }, SPLASH_DISPLAY_TIME.toLong())
    }

    companion object {
        private const val SPLASH_DISPLAY_TIME = 1000
    }
}